ddl/table.go

```diff
@@ -960,6 +961,18 @@ func updateVersionAndTableInfoWithCheck(t *meta.Meta, job *model.Job, tblInfo *m
 // updateVersionAndTableInfo updates the schema version and the table information.
 func updateVersionAndTableInfo(t *meta.Meta, job *model.Job, tblInfo *model.TableInfo, shouldUpdateVer bool) (
 	ver int64, err error) {
+	failpoint.Inject("mockUpdateVersionAndTableInfoErr", func(val failpoint.Value) {
+		switch val.(int) {
+		case 1:
+			failpoint.Return(ver, errors.New("mock update version and tableInfo error"))
```

Comment:
Where do we need 1? It seems only 2 is used.

Reply:
As we talk before, the other DDL has the same problem.  They can reuse this fail point with case 1 which doesn't have to cancel the job directly.